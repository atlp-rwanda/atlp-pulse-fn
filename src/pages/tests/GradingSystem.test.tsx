import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer, { act } from 'react-test-renderer';
import { fireEvent, render, screen } from '@testing-library/react';
import { MockedProvider as ApolloProvider } from '@apollo/client/testing';
import Grading from '../GradingSystem';

describe('Register an Organization', () => {
  it('Renders the register Page', () => {
    const elem = renderer
      .create(
        <MemoryRouter>
          <ApolloProvider>
            <Grading />
          </ApolloProvider>
        </MemoryRouter>,
      )
      .toJSON();
    expect(elem).toMatchSnapshot();
  });

  it('should open add grading system model', () => {
    const removeModelMck = jest.fn();
    const { getByTestId } = render(
      <MemoryRouter>
        <ApolloProvider>
          <Grading />
        </ApolloProvider>
      </MemoryRouter>,
    );
    expect(removeModelMck).toBeCalledTimes(0);
  });

  it('Testing Grading component', async () => {
    render(
      <MemoryRouter>
        <ApolloProvider>
          <Grading />
        </ApolloProvider>
      </MemoryRouter>,
    );

    const gradeName = screen.getByPlaceholderText(
      'Label eg: Name of grading system',
    );
    const gradeDisc = screen.getByPlaceholderText('description');
    const grade = screen.getByPlaceholderText('Grade');
    const gradeBtn = screen.getByTestId('gradeBtn');
    const saveGrade = screen.getByTestId('saveGrade');

    act(() => {
      fireEvent.change(gradeName, {
        target: { value: '' },
      });
    });
    act(() => {
      fireEvent.change(gradeDisc, {
        target: { value: '' },
      });
    });
    act(() => {
      fireEvent.change(grade, {
        target: { value: '' },
      });
    });

    act(() => {
      fireEvent.click(gradeBtn);
      fireEvent.click(saveGrade);
    });

    // test empty input
  });
});
