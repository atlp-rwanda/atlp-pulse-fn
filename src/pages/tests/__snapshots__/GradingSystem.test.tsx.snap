// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register an Organization Renders the register Page 1`] = `
Array [
  <div
    className="h-screen w-screen bg-black fixed top-0 left-0 z-20 bg-opacity-30 backdrop-blur-sm flex items-center justify-center overflow-auto mt-10 p-4 hidden"
  >
    <div
      className="w-screen p-4 pb-8 bg-indigo-100 rounded-lg dark:bg-dark-bg md:w-1/2 xl:w-4/12"
    >
      <div
        className="flex flex-wrap items-center justify-center w-full card-title "
      >
        <h3
          className="w-11/12 text-xl font-bold text-center dark:text-white"
        >
          Delete Grading System
        </h3>
        <hr
          className="w-full my-3 border-gray-400  bg-primary"
        />
      </div>
      <div
        className="card-body"
      >
        <form
          className="px-8 py-3 "
        >
          <div>
            <h2
              className="m-4 text-base text-center dark:text-white"
            >
              reallyRemoveGrading
            </h2>
          </div>
          <div
            className="flex justify-between w-full"
          >
            <button
              className="btn default sm w-[30%] md:w-1/4 text-sm font-sans mx-0 my-2 px-4 border border-current text-violet-500 hover:text-violet-600 hover:bg-violet-600 hover:bg-opacity-5"
              data-testid="delete"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Cancel
            </button>
            <button
              className="btn danger sm w-[30%] md:w-1/4 text-sm font-sans"
              data-testid="confirmDeleteBtn"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Delete
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>,
  <div
    className="h-screen w-screen bg-black fixed top-0 left-0 z-20 bg-opacity-30 backdrop-blur-sm flex items-center justify-center overflow-auto mt-10 p-4 hidden"
  >
    <div
      className="w-screen p-4 pb-8 bg-indigo-100 rounded-lg dark:bg-dark-bg md:w-1/2 xl:w-4/12"
    >
      <div
        className="flex flex-wrap items-center justify-center w-full card-title "
      >
        <h3
          className="w-11/12 text-xl font-bold text-center dark:text-white"
        >
          MakeDefaultTitle
        </h3>
        <hr
          className="w-full my-3 border-gray-400  bg-primary"
        />
      </div>
      <div
        className="card-body"
      >
        <form
          className="px-8 py-3 "
        >
          <div>
            <h2
              className="m-4 text-base text-center dark:text-white"
            >
              reallyMakeDefaultGrading
            </h2>
          </div>
          <div
            className="flex justify-between w-full"
          >
            <button
              className="btn default sm w-[30%] md:w-1/4 text-sm font-sans mx-0 my-2 px-4 border border-current text-violet-500 hover:text-violet-600 hover:bg-violet-600 hover:bg-opacity-5"
              data-testid="delete"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              No
            </button>
            <button
              className="btn default sm w-[30%] md:w-1/4 text-sm font-sans mx-0 my-2 px-4 bg-violet-500 text-white hover:bg-violet-600"
              data-testid="confirmDeleteBtn"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Yes
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>,
  <div
    className="w-screen h-screen bg-black bg-opacity-30 backdrop-blur-sm fixed top-0 left-0 z-20 flex items-center justify-center p-4 hidden"
  >
    <div
      className=" bg-indigo-100 dark:bg-neutral-900 w-full md:w-3/4 xl:w-5/12 rounded-lg p-0 sm:p-4 pb-8 overflow-auto max-h-full"
    >
      <div
        className="card-title w-full flex flex-wrap justify-center items-center "
      >
        <h3
          className="font-bold text-base dark:text-white text-center w-11/12 "
        >
          Add Grading System
        </h3>
      </div>
      <div
        className="card-body"
      >
        <form
          className="py-3 px-4 md:px-8"
          onSubmit={[Function]}
        >
          <div
            className="input my-4 "
          >
            <div
              className="grouped-input flex items-center w-full rounded-md"
            >
              <input
                className="border bg-light-bg dark:bg-neutral-700 bg-opacity-95 border-indigo-200 dark:border-neutral-700 rounded focus:outline-none focus:ring-1 focus:ring-indigo-400 focus:border-indigo-400 dark:ring-0 outline-none px-5 font-sans text-sm py-2 w-full"
                defaultValue=""
                name="name"
                onChange={[Function]}
                placeholder="Label eg: Name of grading system"
                required={true}
                type="text"
              />
            </div>
          </div>
          <div
            className="flex flex-wrap gap-3"
          >
            <label
              className="undefined flex-1 min-w-max"
            >
              <input
                defaultChecked={true}
                onClick={[Function]}
                type="checkbox"
              />
              <div
                className="undefined"
              >
                <div
                  className="undefined"
                >
                   
                </div>
                <span>
                  Add Percentage
                </span>
              </div>
            </label>
            <label
              className="undefined flex-1 min-w-max"
            >
              <input
                defaultChecked={false}
                onClick={[Function]}
                type="checkbox"
              />
              <div
                className="undefined"
              >
                <div
                  className="undefined"
                >
                   
                </div>
                <span>
                  Require Description
                </span>
              </div>
            </label>
          </div>
          <div
            className="my-4"
          >
            <div
              className="box"
            >
              <div>
                <div
                  className="flex flex-col gap-2 p-3 bg-gray-700 bg-opacity-5 dark:bg-neutral-700 dark:bg-opacity-25 rounded-md"
                >
                  <div
                    className="flex flex-col gap-4 "
                  >
                    <input
                      className="rounded-md appearance-none relative block w-full px-4 py-2 border bg-light-bg bg-opacity-95 dark:bg-neutral-700 border-indigo-200 dark:border-neutral-700 placeholder-gray-400 text-gray-900 focus:outline-none focus:ring-1 dark:ring-0 focus:ring-indigo-400 focus:border-indigo-400 focus:z-10 sm:text-sm  dark:text-dark-text-fill"
                      defaultValue=""
                      onChange={[Function]}
                      placeholder="Grade"
                      required={true}
                    />
                    <div
                      className="flex justify-between gap-3 items-center flex-wrap undefined"
                    >
                      <p
                        className="text-center"
                        style={
                          Object {
                            "flex": 1,
                          }
                        }
                      >
                        Percentage
                      </p>
                      <div
                        className="flex justify-between gap-2 items-center flex-wrap max-w-full"
                        style={
                          Object {
                            "flex": 99999,
                          }
                        }
                      >
                        <input
                          className="rounded-md appearance-none relative block max-w-full flex-1 px-3 py-2 border bg-light-bg bg-opacity-95 dark:bg-neutral-700 border-indigo-200 dark:border-neutral-700 placeholder-gray-400 text-gray-900 focus:outline-none focus:ring-1 dark:ring-0 focus:ring-indigo-400 focus:border-indigo-400 focus:z-10 sm:text-sm  dark:text-dark-text-fill"
                          onChange={[Function]}
                          placeholder="Minimum"
                          required={true}
                          type="number"
                        />
                        <input
                          className="rounded-md appearance-none relative block max-w-full flex-1 px-3 py-2 border bg-light-bg bg-opacity-95 dark:bg-neutral-700 border-indigo-200 dark:border-neutral-700 placeholder-gray-400 text-gray-900 focus:outline-none focus:ring-1 dark:ring-0 focus:ring-indigo-400 focus:border-indigo-400 focus:z-10 sm:text-sm  dark:text-dark-text-fill"
                          onChange={[Function]}
                          placeholder="Maximum"
                          required={true}
                          type="number"
                        />
                      </div>
                    </div>
                    <textarea
                      className="w-full h-auto rounded-md appearance-none relative block px-3 py-2 border bg-light-bg bg-opacity-95 dark:bg-neutral-700 border-indigo-200 dark:border-neutral-700 placeholder-gray-400 text-gray-900 focus:outline-none focus:ring-1 dark:ring-0 focus:ring-indigo-400 focus:border-indigo-400 focus:z-10 sm:text-sm  dark:text-dark-text-fill"
                      data-testid="input-element"
                      defaultValue=""
                      onChange={[Function]}
                      placeholder="description"
                      required={false}
                      style={
                        Object {
                          "maxHeight": "auto",
                          "minHeight": "4pc",
                          "resize": "vertical",
                        }
                      }
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="flex justify-between flex-wrap"
          >
            <div
              className="flex justify-start"
            >
              <button
                className="btn default sm text-sm font-sans mx-0 my-2 px-4 bg-violet-500 text-white hover:bg-violet-600"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                Add Grade
              </button>
            </div>
            <div
              className="flex justify-end gap-2 flex-1"
            >
              <button
                className="btn default sm text-sm font-sans mx-0 my-2 px-4 border border-current text-violet-500 hover:text-violet-600 hover:bg-violet-600 hover:bg-opacity-5"
                data-testid="removeModel"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                Cancel
              </button>
              <button
                className="btn default sm text-sm font-sans mx-0 my-2 px-4 bg-violet-500 text-white hover:bg-violet-600"
                data-testid="saveGrade"
                disabled={false}
                type="submit"
              >
                Save
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>,
  <div
    className="flex flex-col"
  >
    <div
      className="flex flex-row"
    >
      <div
        className="w-full"
      >
        <div>
          <div
            className="bg-light-bg dark:bg-dark-frame-bg "
          >
            <div
              className="flex items-left"
            >
              <div
                className="flex flex-wrap gap-2 py-2 pb-8 max-w-full"
              >
                <select
                  className="flex bg-primary rounded-md py-2 px-1 text-white font-medium cursor-pointer max-w-full"
                  onChange={[Function]}
                  value=""
                >
                  <option
                    value=""
                  >
                     ---Select Grading System--- 
                  </option>
                </select>
                <button
                  className="btn primary lg m-0"
                  data-testid="gradeBtn"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                   
                  Grading
                   +
                </button>
              </div>
            </div>
            <div
              className=""
            >
              <div
                className="max-w-full"
              >
                <div
                  className="bg-indigo-100 dark:bg-dark-bg shadow-lg px-5 py-8 rounded-md w-[100%] \\"lg:ml-60 mx-auto\\"} mb-10"
                >
                  <div
                    className="flex items-center justify-between pb-6 "
                  >
                    <div>
                      <h2
                        className="text-xl font-semibold text-gray-800 dark:text-white "
                      >
                        Gradings List
                      </h2>
                      <input
                        className="px-5 py-2 mt-4 font-sans text-xs border border-primary rounded outline-none dark:bg-neutral-600 dark:text-white w-52 md:w-96"
                        defaultValue=""
                        onChange={[Function]}
                        placeholder="Filter"
                      />
                    </div>
                  </div>
                  <div
                    style={
                      Object {
                        "overflowX": "auto",
                      }
                    }
                  >
                    <table
                      className="min-w-full leading-normal"
                      role="table"
                    >
                      <thead>
                        <tr
                          role="row"
                        >
                          <th
                            className=" thead"
                            colSpan={1}
                            onClick={[Function]}
                            role="columnheader"
                            style={
                              Object {
                                "cursor": "pointer",
                              }
                            }
                            title="Toggle SortBy"
                          >
                            Names
                          </th>
                          <th
                            className=" thead"
                            colSpan={1}
                            onClick={[Function]}
                            role="columnheader"
                            style={
                              Object {
                                "cursor": "pointer",
                              }
                            }
                            title="Toggle SortBy"
                          >
                            action
                          </th>
                          <th
                            className=" thead"
                            colSpan={1}
                            onClick={[Function]}
                            role="columnheader"
                            style={
                              Object {
                                "cursor": "pointer",
                              }
                            }
                            title="Toggle SortBy"
                          >
                            More
                          </th>
                        </tr>
                      </thead>
                      <tbody
                        role="rowgroup"
                      />
                    </table>
                    <div>
                      <table
                        className="w-full mt-2"
                      >
                        <tfoot
                          className="w-full py-2"
                        >
                          <tr
                            className="w-full py-2"
                          >
                            <td
                              colSpan={3}
                            >
                              <div
                                className="w-full justify-center flex mx-auto flex-row items-center overflow-x-auto"
                              >
                                <button
                                  className="page mx-2 text-white rounded-circle appearance-none font-bold flex items-center justify-center bg-primary h-[30px] w-[60px] cursor-pointer"
                                  disabled={true}
                                  onClick={[Function]}
                                  type="button"
                                >
                                  <svg
                                    aria-hidden="true"
                                    className="w-5"
                                    fill="none"
                                    fontSize="sm"
                                    stroke="currentColor"
                                    strokeWidth={2}
                                    viewBox="0 0 24 24"
                                    xmlns="http://www.w3.org/2000/svg"
                                  >
                                    <path
                                      d="M11 15l-3-3m0 0l3-3m-3 3h8M3 12a9 9 0 1118 0 9 9 0 01-18 0z"
                                      strokeLinecap="round"
                                      strokeLinejoin="round"
                                    />
                                  </svg>
                                </button>
                                 
                                <button
                                  className="page mx-2 text-white rounded-circle font-bold flex items-center justify-center bg-primary h-[30px] w-[60px] cursor-pointer"
                                  disabled={true}
                                  onClick={[Function]}
                                  type="button"
                                >
                                  <svg
                                    aria-hidden="true"
                                    className="w-5"
                                    fill="none"
                                    fontSize="sm"
                                    stroke="currentColor"
                                    strokeWidth={2}
                                    viewBox="0 0 24 24"
                                    xmlns="http://www.w3.org/2000/svg"
                                  >
                                    <path
                                      d="M13 9l3 3m0 0l-3 3m3-3H8m13 0a9 9 0 11-18 0 9 9 0 0118 0z"
                                      strokeLinecap="round"
                                      strokeLinejoin="round"
                                    />
                                  </svg>
                                </button>
                                 
                                <div
                                  className="flex flex-row justify-center w-full sm:text-[12px] items-center "
                                >
                                  <span
                                    className="inline-block mx-2"
                                  >
                                    Page
                                     
                                    <strong>
                                      1
                                       of
                                       0
                                    </strong>
                                     
                                  </span>
                                  <span
                                    className="inline-block mx-2"
                                  >
                                    | Go to page:
                                     
                                    <input
                                      className=" outline-none border rounded-md appearance-none border-primary dark:bg-primary pl-1"
                                      defaultValue={1}
                                      onChange={[Function]}
                                      style={
                                        Object {
                                          "border": "solid 0.1rem #9e85f5",
                                          "height": "30px",
                                          "paddingLeft": "1.2rem",
                                          "width": "50px",
                                        }
                                      }
                                      type="number"
                                    />
                                  </span>
                                   
                                  <select
                                    className="px-1/2 font-raleway rounded-md border border-primary dark:bg-primary"
                                    onChange={[Function]}
                                    style={
                                      Object {
                                        "border": "solid 0.1rem #9e85f5",
                                        "height": "30px",
                                      }
                                    }
                                    value={3}
                                  >
                                    <option
                                      value={3}
                                    >
                                      Show 
                                      3
                                    </option>
                                    <option
                                      value={5}
                                    >
                                      Show 
                                      5
                                    </option>
                                    <option
                                      value={10}
                                    >
                                      Show 
                                      10
                                    </option>
                                    <option
                                      value={25}
                                    >
                                      Show 
                                      25
                                    </option>
                                    <option
                                      value={50}
                                    >
                                      Show 
                                      50
                                    </option>
                                    <option
                                      value={100}
                                    >
                                      Show 
                                      100
                                    </option>
                                  </select>
                                </div>
                              </div>
                            </td>
                          </tr>
                        </tfoot>
                      </table>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div />
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
